Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift

```diff
@@ -1,20 +1,244 @@
-import Foundation
-class DocumentCaptureViewModel: ObservableObject {
-    private (set) lazy var cameraManager: CameraManageable = CameraManager()
+import SwiftUI
+import Combine
+import CoreVideo
+import AVFoundation
 
-    var navTitle: String {
-        return "Nigeria National ID Card"
+class DocumentCaptureViewModel: ObservableObject, JobSubmittable {
+    enum Position {
+        case front
+        case back
+    }
+    private let rectangleFunnel = RectangleFeaturesFunnel()
+    weak var rectangleDetectionDelegate: RectangleDetectionDelegate?
+    weak var captureResultDelegate: DocumentCaptureResultDelegate?
+    private var displayedRectangleResult: RectangleDetectorResult?
+    private var userId: String
+    private var jobId: String
+    private var document: Document
+    private var currentBuffer: CVPixelBuffer?
+    private (set) var frontImage: UIImage?
+    private var subscribers = Set<AnyCancellable>()
+    private var cameraFeedSubscriber: AnyCancellable?
+    private var captureSubscriber: AnyCancellable?
+    private var captureBothSides: Bool
+    private (set) lazy var cameraManager: CameraManageable = CameraManager(mode: .document)
+    private (set) var postion = Position.front
+    @Published var processingState: ProcessingState? {
+        didSet {
+            switch processingState {
+            case .none:
+                resumeCameraSession()
+            case .endFlow:
+                pauseCameraSession()
+            case .some:
+                pauseCameraSession()
+            }
+        }
     }
 
-    func captureImage() {
+    init(userId: String, jobId: String, document: Document, captureBothSides: Bool) {
+        self.userId = userId
+        self.jobId = jobId
+        self.document = document
+        self.captureBothSides = captureBothSides
+        subscribeToCameraFeed()
+        subscribeToImageCapture()
+    }
 
+    func subscribeToCameraFeed() {
+       cameraFeedSubscriber = cameraManager.sampleBufferPublisher
+            .receive(on: DispatchQueue.global())
```

Comment:
is a global queue the best for this use case (it may be, I'm not too familiar with DispatchQueue)? Assuming this is similar to Kotlin coroutines, I imagine an I/O dedicated queue would be more appropriate